docs/design/multi-table-pipelines/cookies.md

```diff
@@ -0,0 +1,165 @@
+# OpenFlow cookies data
+
+Cookie filed is 64 bits long, it is divided into several bit groups. Some bit fields are system wide i.e. have same meaning for possible cookies, other bit fields are specific for specific cookie type (some subset of cookies).
+
+## System wide cookie fields
+This is set of basic fields that presents in all cookie types.
+
+```
+S00TTTTT TTTT0000 00000000 00000000 00000000 00000000 00000000 00000000
+^                                                                     ^
+`- 63 bit                                                             `- 0 bit
+```
+
+* S - SERVICE_FLAG (1 bit) - OF flow marked with 1 in this bit is a service flow, required for open-kilda to manage service traffic (ISL detection / health check, loops prevention etc)
+* T - TYPE_FIELD (9 bit) - contain cookie type code, this type defines existence and layout all other bit fields in the cookie
+
+
+## Generic cookies (org.openkilda.model.cookie.CookieBase)
+This cookie format was the only existing format into open-kilda from its start. It is not put any restrictions on cookie format and can be used to read "raw" cookie value to determine its actual type.
```

Comment:
`It is not put any restrictions` -> `It doesn't put any restrictions` or `It put no restrictions`